src/main/java/com/cloudbees/hudson/plugins/folder/properties/AuthorizationMatrixProperty.java

```diff
@@ -265,6 +296,20 @@ public Object unmarshal(HierarchicalStreamReader reader,
                 final UnmarshallingContext context) {
             AuthorizationMatrixProperty as = new AuthorizationMatrixProperty();
 
+			String prop = reader.peekNextChild();
```

Comment:
Thanks, good find! It's because there are tabs there, and not spaces. The other AuthorizationMatrixProperty file I copy-pasted this from has a mishmash of tabs and spaces :(.

I'll push a new patchset which uses only spaces.